File: src/main/kotlin/com/xbaimiao/mirai/entity/MessageSubject.java
package com.xbaimiao.mirai.entity;

import com.google.gson.annotations.SerializedName;

public class MessageSubject {
    @SerializedName("id")
    private int id;

    @SerializedName("kind")
    private String kind;

    public int getId() {
        return id;
    }

    public SubjectKind getKind() {
        return SubjectKind.valueOf(kind);
    }

    enum SubjectKind {
        Group,
        Friend
    }
}

File: src/main/kotlin/com/xbaimiao/mirai/entity/MessageSubject.kt
package com.xbaimiao.mirai.entity

import com.google.gson.annotations.SerializedName

class MessageSubject {
    @SerializedName("id")
    var id = 0

    @SerializedName("kind")
    var kind: String? = null

    fun getKind(): SubjectKind{
        return SubjectKind.valueOf(kind!!)
    }

    enum class SubjectKind {
        Group, Friend
    }
}